pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -92,10 +96,81 @@ func (c *APIClient) connect() error {
 	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
 	defer cancel()
 	version, err := c.client.Discovery().Version(ctx)
-	if err == nil {
-		log.Debugf("connected to kubernetes apiserver, version %s", version.GitVersion)
+	if err != nil {
+		return err
 	}
-	return err
+	log.Debugf("connected to kubernetes apiserver, version %s", version.GitVersion)
+	useServiceMapper := config.Datadog.GetBool("service_mapper_disabled")
+	if !useServiceMapper {
+		return nil
+	}
+	c.startServiceMapping()
+	return nil
+}
+
+// ServiceMapperBundle maps the podNames to the serviceNames they are associated with.
+// It is updated by mapServices in services.go
+type ServiceMapperBundle struct {
+	PodNameToServices map[string][]string
+	m                 sync.RWMutex
+}
+
+func newServiceMapperBundle() *ServiceMapperBundle {
+	return &ServiceMapperBundle{
+		PodNameToServices: make(map[string][]string),
+	}
+}
+
+// startServiceMapping is only called once, when we have confirmed we could correctly connect to the API server.
+// The logic here is solely to retrieve Nodes, Pods and Endpoints. The processing part is in mapServices.
+func (c *APIClient) startServiceMapping() {
```

Comment:
This `startServiceMapping` looks solid :+1: 
Do we have an idea if need to stop this ?
I guess the select/case could be used for that.

Reply:
Indeed, I tried to look for where to force stop it, I'm open to suggestions. 
What would you think about a func running after healthy connect, making sure we can query have the proper rights to query the API Server and returns a chan ?